src/track/beatgrid.cpp

```diff
@@ -184,13 +190,49 @@ double BeatGrid::findNthBeat(double dSamples, int n) const {
         n = n + 1;
     }
 
-    double dResult = dClosestBeat + n * m_dBeatLength;
+    double dResult = floor(dClosestBeat + n * m_dBeatLength);
     if (!even(static_cast<int>(dResult))) {
         dResult--;
     }
     return dResult;
```

Comment:
Both implementations of findNthBeats are returning an integer. 
A 32 bit integer is faster on 32 bit CPUs. If you have a close look, we have already a lot of implicit cast from int to double. I would prefer to actually have the static_casts on the caller side. This way, it clarifies that the particular place actually needs decimal places.
It would be easier to identify spot where we have to deal with fractions of samples and round correctly. 
I am afraid, there are some hidden issue because of that. 

For instance, in this place it is not clear for me why you reduce the float beat to an even integer, but we use float beats elsewhere in the code. What would happen if we ditch the rounding here?